### PR TITLE
clear scroll sessions

### PR DESCRIPTION
**Pull request by YuriHechter**

### CODE REVIEW

**Comment by YuriHechter** [comment]
[Clear scroll sessions at index](https://trello.com/c/LtWIgwnI/160-clear-scroll-sessions-at-index)

File: es/index.go [hunk 1]
```diff
@@ -480,6 +481,7 @@ func (index *BaseIndex) RemoveFromIndexQuery(elasticScope elastic.Query) (map[st
 			if _, ok := removed[scrollResult.ResultType]; !ok {
 				removed[scrollResult.ResultType] = make(map[string]bool)
 			}
+			index.esc.ClearScroll(scrollResult.ScrollId)
```
Comment (bbfsdev) [inline es/index.go:484]: Is this the only place where we have scroll that need to be cleared?

File: es/index.go [hunk 2]
```diff
@@ -422,6 +422,7 @@ type ScrollResult struct {
 	MdbUid     string
 	Id         string
 	ResultType string
+	ScrollId   string
 }
 
 func (index *BaseIndex) Scroll(indexName string, elasticScope elastic.Query) ([]ScrollResult, error) {
```
Comment (bbfsdev) [inline es/index.go:428]: where are all the places this is being used?

File: es/index.go (hunk 1 above)
Comment (YuriHechter) [reply es/index.go:484]: Seems to be yes. The only other parts we use scroll in the code is inside GetLattestQueries and GetAllClicks (not in index).

File: es/index.go (hunk 2 above)
Comment (YuriHechter) [reply es/index.go:428]: (1) Line 440: ret = append(ret, ScrollResult{MdbUid: result.MDB_UID, Id: h.Id, ResultType: result.ResultType, ScrollId: searchResult.ScrollId})
(2) Line 484: index.esc.ClearScroll(scrollResult.ScrollId)

**Pull request merged**